File: 1869-longer-contiguous-segments-of-ones-than-zeros/1869-longer-contiguous-segments-of-ones-than-zeros.rb
# @param {String} s
# @return {Boolean}
def check_zero_ones(s)
    zerocount = maxzeros = onecount = maxones = 0
    s.each_char do |char|
        p char
        if char == "1"
            onecount+=1
            maxones=[maxones,onecount].max
            zerocount=0
        elsif char == "0"
            zerocount+=1
            maxzeros=[maxzeros,zerocount].max
            onecount=0
        end
    end
    maxones > maxzeros
end

File: 1869-longer-contiguous-segments-of-ones-than-zeros/1869-longer-contiguous-segments-of-ones-than-zeros.cpp
class Solution {
public:
    bool checkZeroOnes(string s) {
        int zeroCount(0), maxZeros(0), oneCount(0), maxOnes(0);
        for (char chr : s) {
            if (chr == '1') {
                oneCount++;
                maxOnes=max(maxOnes, oneCount);
                zeroCount=0;
            } else if (chr == '0') {
                zeroCount++;
                maxZeros=max(maxZeros, zeroCount);
                oneCount=0;
            }
        }
        return maxOnes > maxZeros;
    }
};